### DISCUSSION TITLE
New collection: infra/aap_utilities

### DISCUSSION THREAD

**Discussion by sean-m-sullivan** [ansible-collections/ansible-inclusion#58] [Suspended]
This Ansible collection includes a number of roles which can be useful for installing and managing AWX or Ansible Automation Platform.

Ansible Galaxy: https://galaxy.ansible.com/infra/aap_utilities
GitHub: https://github.com/redhat-cop/aap_utilities
Issues Tracker: https://github.com/redhat-cop/aap_utilities/issues
GitHub Handles: @sean-m-sullivan , @Tompage1994 , @djdanielsson
Is the collection part of Automation Hub: No, but included in bundle install as validated content
We meet [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst): Yes

For note these collections use shared workflows here: https://github.com/redhat-cop/ansible_collections_tooling

**Comment by Andersson007** [comment]
@sean-m-sullivan hello, thanks for submitting the collections!
I've put am announcement in Bullhorn. As soon as there are volunteers, they'll review the requests

**Comment by Andersson007** [comment]
# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

Note for reviewers: If you don't know how to check any of the points below, please ask maintainers of the collection you're reviewing or a [Steering Committee member](https://docs.ansible.com/ansible/devel/community/steering/community_steering_committee.html#current-steering-committee-members) for clarifications in comments of corresponding inclusion discussion.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] has a policy of releasing, versioning and deprecation announced to contributors and users in some way
  **MUST FIX**: i failed to find any release policies and info that the collection is following semver
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public git repository, releases of the collection are tagged in this repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
- [x] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
- [x] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules satisfy the concept of [idempotency](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency>)
  - [x] modules that only gather information are named `<something>_info`
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [x] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [x] supports Python 2.6 or greater and Python 3.5 or greater. If it does not, read the [full guidelines](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
- [x] only uses the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] documentation, examples, and return sections use FQCNs for the `M(..)` [format macros](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) when referring to modules
- [x] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added
- [x] FQCNs are used for `extends_documentation_fragment:`, unless you are referring to doc_fragments from ansible-core
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [x] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
  **MUST FIX:** I see `redhat.openshift` in README but it's missed in `meta/runtime.yml`
- [x] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
- [x] has changelog, preferably with `changelogs/changelog.yaml`
  **COMMENT**: i see a breaking change was introduced in `2.2.3` which shouldn't happen (if it's just an announcement, there should be a corresponding major version specified)
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files. Temporary files should be added to `.gitignore`.

**Tests:**

Note for reviewers: If you don't know how to check the points below, please ask maintainers of the collection you're reviewing how you can do it.
* In most cases, CI is set up via GitHub Actions.
* Check `.yml` files in the `.github/workflows` directory. There must be at least the `sanity` section under `jobs` containing the `ansible-test sanity` command running against all supported ansible-core versions that must be also listed there, for example, `- stable-2.11, - stable-2.12, - stable-2.13`.
* Check workflow runs by clicking the `Actions` tab in the repository's page - you're interested in `Scheduled` runs, runs against release commits and runs against pull requests.
* If there are no workflows in the `Actions` tab, ask the collection maintainers how CI is implemented.

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it MUST not contain error codes listed [here](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests MUST run against every pull request
- [x] all CI tests MUST run regularly (nightly, or at least once per week)
- [x] sanity tests MUST run against a commit that releases the collection; if they don't pass, the collection won't be released

**Reply by Andersson007** [reply]
@sean-m-sullivan I've just finished my review, see 2 MUST FIXEs and 1 COMMENT, thanks!

**Reply by sean-m-sullivan** [reply]
Release policy we can fix. 
The comment is a good one and looking back it was less a breaking change, then more people really should be aware of the change, so we need to do changelog levels better
The last on the metadata/runtime.yml, I hesitate on including the redhat.openshift, as the collection is dual purpose, For the VM install of AAP and the Openshift install of AAP, I'm going to update the readme in this regard, though.

**Reply by sean-m-sullivan** [reply]
I've updated all 4 of our collections to account for these Fixes.

**Reply by Andersson007** [reply]
cool, thanks, LGTM for inclusion
so now waiting for the second review

**Comment by Andersson007** [comment]
let's hold on a bit, i think we should create a community topic as requirements for roles in collections are not developed yet as was originally brought in https://github.com/ansible-collections/overview/issues/127
